--- a/makefile
+++ b/makefile
@@ -13,10 +13,10 @@
 #      Intel compiler used
 # In the case of ifort 16.0.0 or later
 #######################################
-fort = ifort
-flag = -O2 -mcmodel=medium
-omp = -qopenmp
-nomp = -qopenmp-stubs
+#fort = ifort
+#flag = -O2 -mcmodel=medium
+#omp = -qopenmp
+#nomp = -qopenmp-stubs
 #
 #######################################
 #      Intel compiler used
@@ -33,13 +33,13 @@
 #      GNU Fortran compiler used
 # In the case of gfortran
 #######################################
-#fort = gfortran
-#flag = -O2
-#omp = -fopenmp
-#nomp =
+fort = gfortran
+flag = -O2
+omp = -fopenmp
+nomp =
 #
 program = specx
-#all: $(program) gpd spc
+all: $(program) gpd spc
 objs = \
 source/$(program).o \
 source/spmain.o \
@@ -248,7 +248,7 @@
 source/zroots.o
 
 $(program): $(objs)
-	@$(fort) -o $@ $(flag) $(omp) $(objs) $(libs)
+	$(fort) -o $@ $(flag) $(omp) $(objs) $(libs)
 
 source/specx.o: source/specx.f
 	$(fort) $(flag) -o $(<:.f=.o) -c $<
@@ -261,3 +261,15 @@
 
 spc: source/spc.f
 	$(fort) $(flag) $(omp) -o $@ $<
+
+install: $(program) gpd spc
+	install -d $(DESTDIR)/usr/bin
+	install -d $(DESTDIR)/usr/share/akaikkr/in
+	install -d $(DESTDIR)/usr/share/akaikkr/util
+	install $(program) gpd spc $(DESTDIR)/usr/bin
+	cp -p in/* $(DESTDIR)/usr/share/akaikkr/in
+	cp -p util/* $(DESTDIR)/usr/share/akaikkr/util
+	chmod 644 $(DESTDIR)/usr/share/akaikkr/in/* $(DESTDIR)/usr/share/akaikkr/util/*
+clean:
+	rm -f $(program) gpd spc source/*.o
+
